src/Service/CheckRecaptchaV3.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace OSMAviation\Recaptcha\Service;
+
+/**
+ * Handle sending out and receiving a response to validate the captcha
+ */
+class CheckRecaptchaV3 implements RecaptchaInterface
+{
+
+    /**
+     * Call out to reCAPTCHA and process the response
+     *
+     * @param string $challenge
+     * @param string $response
+     *
+     * @return bool
+     */
+    public function check($challenge, $response)
+    {
+        $parameters = http_build_query([
+            'secret'   => value(app('config')->get('recaptcha.private_key')),
+            'remoteip' => app('request')->getClientIp(),
+            'response' => $response,
+        ]);
+
+        $url           = 'https://www.google.com/recaptcha/api/siteverify?' . $parameters;
+        $checkResponse = null;
+
+        // prefer curl, but fall back to file_get_contents
+        if ('curl' === app('config')->get('recaptcha.driver') && function_exists('curl_version')) {
```

Comment:
Instead of using CURL. Could you use Laravel's HTTP Client? https://laravel.com/docs/8.x/http-client#introduction

Reply:
@tormjens [Done!](https://github.com/osmaviation/recaptcha/pull/1/commits/a4535799c91ffc8f9312838a1713333cbd1860a9)